lib/go-tc/tovalidate/rules.go

```diff
@@ -129,3 +130,33 @@ func IsValidPortNumber(value interface{}) error {
 	}
 	return errors.New("must be a valid port number")
 }
+
+func IsValidIPv6CIDROrAddress(value interface{}) error {
+	switch v := value.(type) {
+	case *string:
+		if v == nil {
+			return nil
+		}
+		ip, _, err := net.ParseCIDR(*v)
+		if err == nil {
+			if ip.To4() == nil {
+				return nil
+			} else {
+				return fmt.Errorf("got IPv4 CIDR, IPv6 expected")
+			}
+		} else {
+			ip := net.ParseIP(*v)
+			if ip != nil {
```

Comment:
You can't combine this nested if?

Reply:
Yeah, we can because we don't care about the difference between an un-parseable string and an IPv4 address.